crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -224,18 +224,18 @@ impl<DB: Database> TransactionsProvider for ProviderFactory<DB> {
         self.provider()?.transaction_id(tx_hash)
     }
 
-    fn transaction_by_id(&self, id: TxNumber) -> Result<Option<TransactionSigned>> {
+    fn transaction_by_id(&self, id: TxNumber) -> Result<Option<TransactionSignedNoHash>> {
         self.provider()?.transaction_by_id(id)
     }
 
-    fn transaction_by_hash(&self, hash: TxHash) -> Result<Option<TransactionSigned>> {
+    fn transaction_by_hash(&self, hash: TxHash) -> Result<Option<TransactionSignedNoHash>> {
```

Comment:
I can see how this is useful for perf

I'd rather have this as separate trait functions like `transaction_by_id_no_hash` instead and keep returning `TransactionSigned`

Reply:
Cool il do that then less changes and fine my end 👍

Reply:
i'm not really sure about it though, it hides the performance cost. Can we just add to the doc of the `databaseprovider` implementation, that it incurs that cost at least?

Reply:
let me know what you guys want to do.. @mattsse @joshieDo

Reply:
> 'm not really sure about it though

about what exactly?

Reply:
added a note on it

Reply:
> about returning TransactionSigned per default on functions that need to calculate the hash. the calculation cost is kinda hidden imo

right, I was strictly looking at this from rpc pov where we always need this